kafka/sink/pkg/adapter/adapter.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adapter
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/Shopify/sarama"
+	cloudeventskafka "github.com/cloudevents/sdk-go/protocol/kafka_sarama/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cloudeventshttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+)
+
+type envConfig struct {
+	adapter.EnvConfig
+
+	// Port on which to listen for cloudevents
+	Port string `envconfig:"PORT" default:"8080"`
+
+	// KafkaServer URL to connect to the Kafka server.
+	KafkaServer string `envconfig:"KAFKA_SERVER" required:"true"`
+
+	// Topic to publish cloudevents to.
+	Topic string `envconfig:"KAFKA_TOPIC" required:"true"`
+}
+
+// NewEnvConfig function reads env variables defined in envConfig structure and
+// returns accessor interface
+func NewEnvConfig() adapter.EnvConfigAccessor {
+	return &envConfig{}
+}
+
+// kafkaSinkAdapter acts as Sink that sends incoming CloudEvents to Kafka topic
+type kafkaSinkAdapter struct {
+	logger *zap.SugaredLogger
+	client cloudevents.Client
+	//source string
+
+	port        string
+	kafkaServer string
+	topic       string
+}
+
+// NewAdapter returns the instance of gitHubReceiveAdapter that implements adapter.Adapter interface
+func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {
```

Comment:
Why you create a source like adapter if then you don't use the provided client?

Reply:
That is to structure it as receive adapter for future PR that add controller support and KafkaSink CR